contracts/utils/cryptography/MerkleProof.sol

```diff
@@ -23,21 +23,36 @@ library MerkleProof {
         bytes32 root,
         bytes32 leaf
     ) internal pure returns (bool) {
-        bytes32 computedHash = leaf;
+        // Check if the computed hash (root) is equal to the provided root
+        (bytes32 computedHash, ) = processProof(proof, leaf);
+        return computedHash == root;
+    }
 
+    /**
+     * @dev Returns the rebuild hash of node obtained by traversing a Merklee
```

Comment:
Can you document the `index` value that is returned? I don't really understand it either. :smiley:

Reply:
I'm mostly sure (need a proof) of uniqueness, but I have counter-examples showing non-ordering and non-sequentiality (mostly because of the "branches" being has order rather than branch direction)

Reply:
It can be useful to mark a leaf-proof combo as consumed in case a leaf hash is present multiple times in the tree.